src/random.h

```diff
@@ -67,7 +67,17 @@
  * Thread-safe.
  */
 void GetRandBytes(unsigned char* buf, int num) noexcept;
+/** Generate a random integer in the range [0..range). Precondition: range > 0 */
 uint64_t GetRand(uint64_t nMax) noexcept;
+// The following GetRand*(std::chrono::*) MUST NOT be templated on the type of
+// the duration to prevent unsafe use at the call site. If it was templated, it
+// would be impossible for the compiler to select the right template. E.g.
+// assuming that a GetRand<T> exists for all duration types, then `foo` will
+// always be set to a constant, not a random duration.
```

Comment:
I don't think this is true? its just random in relation to the duration type of the input. 

eg. `std::chrono::seconds foo = GetRand(std::chrono::hours{5})`
might return 3600, 7200, 10800 etc.

Reply:
Any suggestions to improve the wording? Something with "quantized", maybe?

Reply:
think precision loss captures it better, esp with example. attempted in other comment thread.